### PR TITLE
Clarify encapsulated.

### LINKED ISSUE #3270: Clarification about encapsulated short class definition
Section 4.5.1 says:
> An exception to the above is that if the short class definition is declared as encapsulated, then the modifiers follow the rules for encapsulated classes and cannot be looked up in the enclosing scope.

Which only provides a constraints for the modifiers.

Does that mean that the following is valid:
```
package P
  model A
    Real x = 1;
  end A;

  encapsulated model B = A;
end P;
```

### PR DESCRIPTION
**Pull request by HansOlsson**
Change grammar for two reasons:
* It is type-specifier not IDENT
* For encapsulated we really need the leading "." which is part of type-specifier.

Closes #3270

### CODE REVIEW

File: chapters/classes.tex [hunk 1]
```diff
@@ -644,20 +644,20 @@ \subsection{Short Class Definitions}\label{short-class-definitions}
 
 A class definition of the form
 \begin{lstlisting}[language=modelica]
-class IDENT1 = IDENT2 class-modification;
+class IDENT1 = type-specifier class-modification;
 \end{lstlisting}
 
-is identical, except that IDENT2 may be replaceable and for the lexical
+is identical, except that name may be replaceable and for the lexical
```
Comment (perost) [inline chapters/classes.tex:650]: "name" is a bit ambiguous here I think, maybe use e.g. "the base class" instead?

Comment (HansOlsson) [reply chapters/classes.tex]: I changed to type-specifier, which is unambiguous, and added (the base-class) in parenthesis.

File: chapters/classes.tex [hunk 2]
```diff
@@ -644,20 +644,20 @@ \subsection{Short Class Definitions}\label{short-class-definitions}
 
 A class definition of the form
 \begin{lstlisting}[language=modelica]
-class IDENT1 = IDENT2 class-modification;
+class IDENT1 = type-specifier class-modification;
 \end{lstlisting}
```
Comment (henrikt-ma) [inline chapters/classes.tex:649]: This empty line causes a paragraph break that shouldn't be here:
```suggestion
```

File: chapters/classes.tex [hunk 3]
```diff
@@ -644,20 +644,20 @@ \subsection{Short Class Definitions}\label{short-class-definitions}
...
 
-is identical, except that IDENT2 may be replaceable and for the lexical
+is identical, except that \lstinline!type-specifier! (the base-class) may be replaceable and for the lexical
 scope of modifiers, where the short class definition does not introduce
 an additional lexical scope for modifiers, to the longer form
```
Comment (henrikt-ma) [inline chapters/classes.tex:653]: Empty line causing bad paragraph break:
```suggestion
```

(If one really wants a paragraph break at a place like this, it needs to be explained why in a comment.)

File: chapters/classes.tex [hunk 4]
```diff
@@ -644,20 +644,18 @@ \subsection{Short Class Definitions}\label{short-class-definitions}
 
 A class definition of the form
 \begin{lstlisting}[language=modelica]
-class IDENT1 = IDENT2 class-modification;
+class IDENT1 = type-specifier class-modification;
 \end{lstlisting}
-
-is identical, except that IDENT2 may be replaceable and for the lexical
+is identical, except that \lstinline!type-specifier! (the base-class) may be replaceable and for the lexical
```
Comment (eshmoylova) [inline chapters/classes.tex:649]: This sentence is broken on multiple lines.

**Comment by henrikt-ma** [comment]
@HansOlsson, maybe you could take a look at the merge conflicts while we wait for the approving review?

**Comment by HansOlsson** [comment]
> @HansOlsson, maybe you could take a look at the merge conflicts while we wait for the approving review?

Have done. However, since repeated merging can be messy I generally postpone that until the end.

**Review by qlambert-pro — APPROVED** [review]
lgtm

**Pull request merged**